GameServerLib/Items/Inventory.cs

```diff
@@ -130,7 +143,13 @@ private IItem AddStackingItem(IItemData item)
             }
             return AddNewItem(item);
         }
+        public void RemoveStackingItem(Game game, string itemSpellName, IObjAiBase owner)
+        {
+            IItem item = GetItem(itemSpellName);
 
+            game.PacketNotifier.NotifyRemoveItem(owner, GetItemSlot(item), (byte)(item.StackCount - 1));
```

Comment:
PacketNotifier and Game class shouldn't be in Inventory. The packets should be sent after the inventory has been updated in each event (and there are only few: Buy, sell and use).

Reply:
Something you could do is add a _toRemove boolean variable to Item.cs, then inside ObjectManager.cs, check each unit with an inventory for removed items, and call the packets there.

Reply:
resolved